### PR TITLE
[Storage] Clear token on signout

### PR DESCRIPTION
**Pull request by annakocheshkova**
* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Delete user's tokens in the Data SDK module.

## Related PRs or issues

List related PRs and other issues.

## Misc

Related Work Item: https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/57722

### CODE REVIEW

**Comment by annakocheshkova** [comment]
New methods test coverage remained 100%. Ignore the CI.

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java [hunk 1]
```diff
@@ -311,6 +318,27 @@ public void deleteEndToEnd() {
         // TODO: assert error is null and document is null
     }
 
+    @Test
+    public void tokenClearedOnSignOut() {
+        Set<String> partitionNames = new HashSet<>();
```
Comment (YiMysty) [inline sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: Add comments /* when */ /* Call */  /* verify */

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java [hunk 2]
```diff
@@ -88,4 +88,17 @@ public synchronized void removeCachedToken(String partitionName) {
         SharedPreferencesManager.putStringSet(Constants.PARTITION_NAMES, partitionNamesSet);
         SharedPreferencesManager.remove(partitionName);
     }
+
+    /**
+     * Remove all the cached access tokens for all partition names.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public synchronized void removeAllCachedTokens() {
+        Set<String> partitionNamesSet = getPartitionNames();
```
Comment (guperrot) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java]: If user signs out, we must keep tokens related to read-only partitions

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java (hunk 1 above)
Comment (jaelim-ms) [reply sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: Well, we don't really care in Android code. 😄

Comment (annakocheshkova) [reply sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: @YiMysty there are different code style conventions in Android SDK :)

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java (hunk 2 above)
Comment (annakocheshkova) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java]: @guperrot how can we determine from the SDK side whetehr the partition name is read-only?

File: sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java (hunk 1 above)
Comment (jwallra) [reply sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java]: I've made these comments in the past, I guess we don't have a hard requirement for it in android.  Sorry about that @YiMysty 

Although I do think they're a good idea and if you want to add them that would be great :)

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java (hunk 2 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java]: I think it's just comparing the name to Constants.READONLY, is that correct @YiMysty ?

Comment (YiMysty) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java]: yup!

**Pull request merged**